src/Components/Components/src/RenderTree/Renderer.cs

```diff
@@ -801,5 +851,26 @@ public void Dispose()
         {
             Dispose(disposing: true);
         }
+
+        /// <inheritdoc />
+        public async ValueTask DisposeAsync()
+        {
+            if (_disposeTask != null)
+            {
+                await _disposeTask;
+            }
+            else
+            {
+                Dispose();
+                if (_disposeTask != null)
```

Comment:
Does this mean `Dispose()` does not dispose these async components?

Reply:
Dispose disposes async components, but doesn't wait for them to finish

Reply:
We only call dispose async in our implementations, but I can't "remove" Dispose since is part of the public API

Reply:
What do you feel about making a blocking call in `Dispose`? `Dispose() => DisposeAsync().AsTask().GetAwaiter().GetResult()`? `DisposeAsync can call `Dispose(diposing: true)` directly

Reply:
I'd rather remove it completely in a future release.